apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+      'click', function onClick(event) {
+        event.stopPropagation();
+      }
+    );
+
+    navigator.mozContacts.addEventListener(
+      'contactchange',
+      this.updateHeaderData.bind(this)
+    );
+
+    this.recipientSuggestions.addEventListener(
+      'click',
+      this.onRecipientSuggestionClick.bind(this)
+    );
+
+    MessageManager.on('message-sending', this.onMessageSending.bind(this));
+    MessageManager.on('message-sent', this.onMessageSent.bind(this));
+    MessageManager.on('message-received', this.onMessageReceived.bind(this));
+    MessageManager.on(
+      'message-failed-to-send',
+      this.onMessageFailed.bind(this)
+    );
+    MessageManager.on('message-delivered', this.onDeliverySuccess.bind(this));
+    MessageManager.on('message-read', this.onReadSuccess.bind(this));
+
+    this.tmpl = templateIds.reduce(function(tmpls, name) {
+      tmpls[Utils.camelCase(name)] =
+        Template('messages-' + name + '-tmpl');
+      return tmpls;
+    }, {});
+
+    Compose.init('messages-compose-form');
+
+    // In case of input, we have to resize the input following UX Specs.
+    Compose.on('input', this.messageComposerInputHandler.bind(this));
+    Compose.on('subject-change', this.onSubjectChange.bind(this));
+    Compose.on('segmentinfochange', this.onSegmentInfoChange.bind(this));
+
+    // Assimilations
+    // -------------------------------------------------
+    // If the user manually types a recipient number
+    // into the recipients list and does not "accept" it
+    // via <ENTER> or ";", but proceeds to either
+    // the message or attachment options, attempt to
+    // gather those stranded recipients and assimilate them.
+    //
+    // Previously, an approach using the "blur" event on
+    // the Recipients' "messages-to-field" element was used,
+    // however the to-field will frequently lose "focus"
+    // to any of its recipient children. If we assimilate on
+    // to-field blur, the result is entirely unusable:
+    //
+    //  1. Focus will jump from the recipient input to the
+    //      message input
+    //  2. 1 or 2 characters may remain in the recipient
+    //      editable, which will be "assimilated"
+    //  3. If a user has made it past 1 & 2, any attempts to
+    //      select a contact from contact search results
+    //      will also jump focus to the message input field
+    //
+    // So we assimilate recipients if user starts to interact with Composer
+    Compose.on('interact', this.assimilateRecipients.bind(this));
+
+    this.container.addEventListener(
+      'click', this.assimilateRecipients.bind(this)
+    );
+
+    this.multiSimActionButton = null;
+
+    this.timeouts.update = null;
+
+    this.shouldChangePanelNextEvent = false;
+
+    this.showErrorInFailedEvent = '';
+
+    // Bound methods to be detachables
+    this.onMessageTypeChange = this.onMessageTypeChange.bind(this);
+  },
+
+  onVisibilityChange: function conv_onVisibilityChange(e) {
```

Comment:
As I mentioned above - this method can be split into two: Conversation and NewMessage specific.